### PR TITLE
Prettier v2

### PR DESCRIPTION
**Pull request by zoran995**
### What this PR does

Enables fix of #5420

This is pretty big change for terriajs codebase but version 2 of prettier came out 2 and half years and it would be pretty good to upgrade to it. It also allows us to use import just type and some other modern syntax that was erroring on old versions.

If decided to merge it should be done the same way as the last time, by creating two new tags pre-prettier-v2 and post-prettier-v2

### Checklist

-   ~[ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)~
-   ~[ ] I've updated relevant documentation in `doc/`.~
-   [ ] I've updated CHANGES.md with what I changed.
-   ~[ ] I've provided instructions in the PR description on how to test this PR.~

### CODE REVIEW

File: buildprocess/generateCatalogIndex.ts [hunk 1]
```diff
@@ -136,13 +136,15 @@ export default async function generateCatalogIndex(
...
+            const result = await (
+              member as ReferenceMixin.Instance
+            ).loadReference();
             result.logError(`FAILED to load Reference ${name} (${path})`);
             result.pushErrorTo(
               errors,
               `FAILED to load Reference ${name} (${path})`
             );
-            result.catchError(e => console.error(e.toError().message));
+            result.catchError((e) => console.error(e.toError().message));
```
Comment (nf-s) [inline buildprocess/generateCatalogIndex.ts:147]: Is it possible to configure prettier to keep single argument anon functions without brackets?

**Review by nf-s — CHANGES_REQUESTED** [review]
Thanks @zoran995 !
Just one comment - otherwise looks good!

File: buildprocess/generateCatalogIndex.ts (hunk 1 above)
Comment (zoran995) [reply buildprocess/generateCatalogIndex.ts:147]: Yes setting `"arrowParens": "avoid"` in `.prettierrc` will do that, I personally prefer to always add parens as it is easier to add type annotation, another param, consistency and bunch of other options, I also believe most of the developers are used to this new syntax. but in the end prettier handles them so perfectly fine with both

https://prettier.io/blog/2020/03/21/2.0.0.html#change-default-value-for-arrowparens-to-always-7430httpsgithubcomprettierprettierpull7430-by-kachkaevhttpsgithubcomkachkaev

Comment (nf-s) [reply buildprocess/generateCatalogIndex.ts:147]: Yep fair enough, I think you're right.

**Comment by tephenavies** [comment]
We should extend this to also apply prettier to more files in the repo (e.g. gulpfile.js, everything in buildprocess, markdown files). Maybe even change our prettierignore file to ignore certain files rather than only include some directories.

**Comment by zoran995** [comment]
@steve9164 I have updated prettierignore to include more files

**Comment by tephenavies** [comment]
That's better. I think we should also format:
* `*.yml` files
* `architecture`
* plenty of things in `wwwroot/` (probably most things that aren't `.gitignore`d, third party code or generated files that are never partially modified)
* `.github`

I was thinking of changing `.prettierignore` to almost match `.gitignore`, plus add a few extra lines.

`.prettierignore`:
```.gitignore

# =====================
# From .gitignore. Update this list when .gitignore changes

docs/
node_modules/
.idea/
build/
error.log
output.log
npm-debug.log
.DS_Store
terria.lib.js
terriajs-server.log
terriajs.pid
/wwwroot/doc/
/wwwroot/schema/
/wwwroot/user-guide/
/coverage/
package-lock.json
*.swp
dist
.history/

# Catalog index generation files
catalog-index*.json

# End of .gitignore copy
# =====================

# These are auto-generated by css-modules-typescript-loader
**/*.scss.d.ts

# Generated files
/wwwroot/data/regionids

/wwwroot/third_party

# Reignore some files
doc/acknowledgements/attributions.md

```

And change the package.json scripts to  `"prettier --write ."` and `"prettier --check ."`. That'll circumvent the problem where prettier is trying to format files it doesn't recognise when the ignore file is reversed.

**Comment by tephenavies** [comment]
Merging on 22nd August.

**Pull request closed without merging**